### PR TITLE
Add support to enable/disable the event handler

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
## Purpose
* $subject to provide the capability to enable or disable the event handler through the configuration.

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/handler/UserInvitationEventHandler.java [hunk 1]
```diff
@@ -51,6 +58,22 @@ public int getPriority(MessageContext messageContext) {
         return 100;
     }
 
+    public boolean canHandle(MessageContext messageContext) throws IdentityRuntimeException {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/handler/UserInvitationEventHandler.java:61]: Why we need to overide the canHandle method. Can't we have the event handler enabled check as the first line of the event handler method.

Comment (ShanChathusanda93) [reply components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/handler/UserInvitationEventHandler.java:61]: We can, but the process of checking whether the handler can handle the task is done through this canHandle method. Before calling the handle method, the canHandle will be executed. If the canHandle returnes true, then only the handle method is executing.

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/handler/UserInvitationEventHandler.java:61]: The canhandle method implementation looks verifying the subscription. Will it cause any issues by overriding that logic.
```[[event_handler]]name= "UserInvitationEventHandler"
subscriptions =["POST_ADD_ORGANIZATION_USER_INVITATION"]```

Comment (ShanChathusanda93) [reply components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/handler/UserInvitationEventHandler.java:61]: yes, [1] is the super implementation. There are the subscription is checked. In this implementation also the subscription is checked from the following code.
```
String eventName = ((IdentityEventMessageContext) messageContext).getEvent().getEventName();
if (EVENT_NAME_POST_ADD_INVITATION.equals(eventName)) {
    if (LOG.isDebugEnabled()) {
        LOG.debug("canHandle() returning True for the event: " + eventName);
    }
    return true;
}
```

[1] https://github.com/wso2/carbon-identity-framework/blob/b4154be22c22e22af01b16c030b513cf61c01836/components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java#L45

**Pull request merged**